stingray/lightcurve.py

```diff
@@ -1001,7 +1001,7 @@ def join(self, other):
         gti = join_gtis(self.gti, other.gti)
 
         lc_new = Lightcurve(new_time, new_counts, err=new_counts_err, gti=gti,
-                            mjdref=self.mjdref, dt=self.dt)
+                            mjdref=self.mjdref, dt=self.dt, skip_checks=False)
```

Comment:
Rather than defaulting to `False` in this lightcurve creation (which is indeed a good default, good intuition here!), I would allow users to skip checks if they know what they're doing, directly when calling `join()`. Basically, adding a `skip_checks` keyword to the `join` method